PDK_Abstraction/FinFET14nm_Mock_PDK/FinFET_Mock_PDK_Abstraction.json

```diff
@@ -8,8 +8,10 @@
 "Direction": "V",
 "Pitch": 80,
 "Width": 14,
-"MinL" : 60,
-"MaxL" : null
+"MinL": 60,
+"MaxL": null,
+"EndToEnd": null,
+"Offset": 40
```

Comment:
Sorry meant to start a conversation on this subject:

> We used the "Offset" to have a different starting point for "Poly" and "M1" or if we do not want to start a layer from the origin. And what layers it applies to? : Metal layers, poly, and Fin. @just want to make sure there is a 1:1 correlation between the "EnclosureGrid" offset and the JSON Offset you are proposing. The offset will affect the "CenterlineGrid" and "EnclosureGrid" offset. E.g. M1 "stopingPoint" is related to M2 pitch and offset.

Starting a review to discuss this further. 

I don't think adding "Offset" to the JSON for this purpose hurts... But am not clear if it helps either if we are only talking about the fact that M1 lines should be 0.5 poly pitches away from the Poly (This can be done when instantiating the device). If we have metal to metal offsets, then we definitely need this parameter. I think @stevenmburns has seen something like this before?

If we do add it to the JSON, then we should definitely use it for DefaultCanvas. But you would likely need to rewrite a couple of tests as well.

You also mentioned something about M1 offset = 0 vs. offset = pitch // 2. This is purely a stylistic choice. pitch // 2 just makes sure that gridpoint (0, 1) has a positive physical coordinate.  It could have been hardcoded to zero as well.

Reply:
I can modify the existing tests; 
I think the "Offset" will help when we abstract other PDKs or more complex rules.

Reply:
Please go ahead in that case. Let me know if you want me to take a look !